easybuild/tools/filetools.py

```diff
@@ -460,26 +463,49 @@ def download_file(filename, url, path, forced=False):
     attempt_cnt = 0
 
     # use custom HTTP header
-    url_req = urllib2.Request(url, headers={'User-Agent': 'EasyBuild',  "Accept" : "*/*"})
+    headers = {'User-Agent': 'EasyBuild', 'Accept': '*/*'}
+    if not HAVE_REQUESTS:
+        url_req = urllib2.Request(url, headers=headers)
 
     while not downloaded and attempt_cnt < max_attempts:
         try:
-            # urllib2 does the right thing for http proxy setups, urllib does not!
-            url_fd = urllib2.urlopen(url_req, timeout=timeout)
-            _log.debug('response code for given url %s: %s' % (url, url_fd.getcode()))
+            if HAVE_REQUESTS:
+                url_req = requests.get(url, headers=headers, stream=True, timeout=timeout)
+                status_code = url_req.status_code
+                url_req.raise_for_status()
+                url_fd = url_req.raw
+                url_fd.decode_content = True
+            else:
+                # urllib2 does the right thing for http proxy setups, urllib does not!
+                url_fd = urllib2.urlopen(url_req, timeout=timeout)
+                status_code = url_fd.getcode()
+            _log.debug('response code for given url %s: %s' % (url, status_code))
             write_file(path, url_fd.read(), forced=forced, backup=True)
             _log.info("Downloaded file %s from url %s to %s" % (filename, url, path))
             downloaded = True
             url_fd.close()
-        except urllib2.HTTPError as err:
-            if 400 <= err.code <= 499:
-                _log.warning("URL %s was not found (HTTP response code %s), not trying again" % (url, err.code))
+        except HTTPError as err:
+            if not HAVE_REQUESTS:
+                status_code = err.code
+            if 400 <= status_code <= 499:
+                _log.warning("URL %s was not found (HTTP response code %s), not trying again" % (url, status_code))
                 break
             else:
                 _log.warning("HTTPError occurred while trying to download %s to %s: %s" % (url, path, err))
                 attempt_cnt += 1
         except IOError as err:
             _log.warning("IOError occurred while trying to download %s to %s: %s" % (url, path, err))
+            error_re = re.compile(r"<urlopen error \[Errno 1\] _ssl.c:.*: error:.*:"
+                                  "SSL routines:SSL23_GET_SERVER_HELLO:sslv3 alert handshake failure>")
+            if error_re.match(str(err)):
```

Comment:
This construct is used to switch to trying with `requests`, but it's a bit messy, especially with the inline `import` statements...
I think we need a `use_requests = True` or maybe `broken_urrlib = True` here which can be checked above?

```python
# for backward compatibility, and to avoid relying on 3rd party Python library 'requests'
use_requests = False
while not downloaded and attempt_cnt < max_attempts:
    if use_requests:
        if not HAVE_REQUESTS:
           raise EasyBuildError("...")
        # do requests...
    else:
        # do what we do now
        try:
            ...
        except IOError as err:
            if error_re.match(str(err)):
                use_requests = True